### PR TITLE
Sum Of Multiples mentoring notes for Python

### PR DESCRIPTION
**Pull request by JohnCrickett**
Adding mentoring notes for Sum of Multiples based on solutions and issues seen mentoring it.

### CODE REVIEW

**Comment by JohnCrickett** [comment]
@yawpitch good suggestion, thanks. I've added that to the notes.

File: tracks/python/exercises/sum-of-multiples/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+When limit is very large and values in multiples are high in the range, there could be a lot of computational time wasted on unnecessary division, so a collection-based approach may be more suitable:
+
+```python
+def sum_of_multiples(limit, multiples):
+    return sum({n for m in multiples for n in range(m, limit, m)})
+```
+
+### Common Suggestions
+ - It's tempting to write a complex one liner for this, but that should be discouraged in favour of
+   PEP8 complaint solutions for readability and maintainability.
```
Comment (rpottsoh) [inline tracks/python/exercises/sum-of-multiples/mentoring.md]:
```suggestion
   PEP8 compliant solutions for readability and maintainability.
```

File: tracks/python/exercises/sum-of-multiples/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+### Reasonable solutions
+
+This solution deliberately shows the two steps seperately:
```
Comment (rpottsoh) [inline tracks/python/exercises/sum-of-multiples/mentoring.md]:
```suggestion
This solution deliberately shows the two steps separately:
```

File: tracks/python/exercises/sum-of-multiples/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            if number % multiple == 0:
+                return True
+        return False
+
+    numbers = filter(is_multiple, range(limit))
+
+    return sum(numbers)
+```
+
+The solution below shows a terser approach:
```
Comment (rpottsoh) [inline tracks/python/exercises/sum-of-multiples/mentoring.md]:
```suggestion
The solution below shows a more terse approach:
```

**Comment by rpottsoh** [comment]
This PR should be squashed when it is merged.

**Pull request merged**